signer/signer.go

```diff
@@ -130,13 +130,17 @@ type Configuration struct {
 	SignerOpts crypto.SignerOpts `json:"signer_opts,omitempty" yaml:"signeropts,omitempty"`
 
 	isHsmAvailable bool
-	hsmCtx         *pkcs11.Ctx
+	Hsm HSM
 }
 
 // InitHSM indicates that an HSM has been initialized
 func (cfg *Configuration) InitHSM(ctx *pkcs11.Ctx) {
 	cfg.isHsmAvailable = true
-	cfg.hsmCtx = ctx
+	cfg.Hsm = &AWSHSM{
```

Comment:
I like how you've avoided the siren song I could not: moving these methods out of Configuration entirely. This is probably wiser.

This is instantiation is pretty deep in here. What would this and its calling code look like when we added the GCP version?

Could we maybe have `InitHSM` take the interface type instead of `*pkcs11.Ctx` and do whatever env var or autograph config file checking we need to in `main.go`?

I have two strands of feedback here in one line of code: We should probably have a version that works in GCP in this patch so that we can know we've got the right joints exposed, and we probably want the switching between the versions of these types to happen at boot time in `main.go` somewhere.

Reply:
Also, thanks for your patience on the feedback. I was mulling this over hours ago and got distracted helping with some other problems.

Reply:
(Also, that "do the choosing between AWSHSM and GCPHSM in main.go" will def push me into the camp of "make constructor NewAWSHSM and NewGCPHSM funcs to hide their GenericHSM wrapping details"

Reply:
> Could we maybe have InitHSM take the HSM interface type instead of *pkcs11.Ctx and do whatever env var or autograph config file checking we need to in main.go?

Yeah, that sounds like a great idea.

> I have two strands of feedback here in one line of code: We should probably have a version that works in GCP in this patch so that we can know we've got the right joints exposed

We're getting to the point where we have recursive dependencies between #987 and this :stuck_out_tongue:. Part of the reason I did this refactoring now was so that I could add the GCP MakeKey implementation as part of that PR. It sounds like I should just roll that, plus this, plus the `GCPHSM` implementation into one PR for the (good) testing reasons you mentioned?

> we probably want the switching between the versions of these types to happen at boot time in main.go somewhere.

Will do.

Reply:
Nah, you're right. I'm being a bother. We can leave these separate